### PR TITLE
cherry-pick #7595 to v1.67.x branch

### PR DESCRIPTION
**Pull request by arjan-bal**
Original PR: #7595 

RELEASE NOTES: N/A

### CODE REVIEW

**Comment by purnesh42H** [comment]
I think you still have to add the same release notes? LGTM otherwise

**Review by purnesh42H — APPROVED** [review]

**Comment by arjan-bal** [comment]
> I think you still have to add the same release notes? LGTM otherwise

Since the PR is already present in the latest release (1.66.2), I don't think we need to mention it in the release notes for 1.67. Confirming with other maintainers.

**Pull request merged**
